addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/Log4ShellScanRule.java

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.addon.oast.ExtensionOast;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+public class Log4ShellScanRule extends AbstractAppParamPlugin {
+
+    private static final Logger LOGGER = LogManager.getLogger(Log4ShellScanRule.class);
+    private static final String PREFIX = "ascanalpha.log4shell.";
+    private static final String ATTACK_PATTERN = "${jndi:ldap://{0}/}";
+
+    @Override
+    public int getId() {
+        return 40043;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(PREFIX + "name");
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        return technologies.includesAny(Tech.JAVA, Tech.SPRING);
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.MISC;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(PREFIX + "refs");
+    }
+
+    @Override
+    public int getRisk() {
+        return Alert.RISK_HIGH;
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        Map<String, String> alertTags =
+                new HashMap<>(
+                        CommonAlertTag.toMap(
+                                CommonAlertTag.OWASP_2021_A06_VULN_COMP,
+                                CommonAlertTag.OWASP_2017_A09_VULN_COMP,
+                                CommonAlertTag.WSTG_V42_INPV_11_CODE_INJ));
+        alertTags.put(ExtensionOast.OAST_ALERT_TAG_KEY, ExtensionOast.OAST_ALERT_TAG_VALUE);
+        return alertTags;
+    }
+
+    @Override
+    public int getCweId() {
+        return 117;
+    }
+
+    @Override
+    public int getWascId() {
+        return 20;
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        try {
+            ExtensionOast extOast =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionOast.class);
+            if (extOast == null) {
+                return;
+            }
+            if (extOast.getActiveScanOastService() != null) {
```

Comment:
This could be done in the init and skip the rule (e.g. `UsernameEnumerationScanRule`).